sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -392,8 +392,44 @@ case class NewInstance(
     childrenResolved && !needOuterPointer
   }
 
-  override def eval(input: InternalRow): Any =
-    throw new UnsupportedOperationException("Only code-generated evaluation is supported.")
+  private lazy val constructor: (Seq[AnyRef]) => Any = {
+    val paramTypes = arguments.map { expr =>
+      CallMethodViaReflection.typeMapping.getOrElse(expr.dataType,
+        Seq(expr.dataType.asInstanceOf[ObjectType].cls))
+    }
+    val findConstructor = (types: Seq[Seq[Class[_]]]) => {
+      val constructorOption = cls.getConstructors.find { c =>
```

Comment:
Can we directly call `cls.getConstructor` with required `Class` to get it?

Reply:
ok, I'll re-check.

Reply:
How does this compare to JVM resolution rules? The code generated version will follow those.